### PR TITLE
[GEOT-7077] ClientProperty of a top element not showing

### PR DESCRIPTION
**Pull request by taba90**
[GEOT-7077](https://osgeo-org.atlassian.net/browse/GEOT-7077) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Details on Jira ticket.

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/data/complex/ClientPropertiesTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        PropertyName clientProp2 = ff.property("@staticProp");
+
+        Query query = new Query();
+        query.setProperties(Query.ALL_PROPERTIES);
+        query.setFilter(Filter.INCLUDE);
+        PropertyName namePn = ff.property("gml:name");
+        FeatureCollection<FeatureType, Feature> mfCollection = mfSource.getFeatures(query);
+
+        try (FeatureIterator iterator = mfCollection.features()) {
+            int i = 0;
```
Comment (aaime) [inline modules/extension/app-schema/app-schema/src/test/java/org/geotools/data/complex/ClientPropertiesTest.java:91]: i being... featureCount? Maybe rename the variable accordingly.

**Review by aaime — CHANGES_REQUESTED** [review]
Overall looks good, minor suggestions below.

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/data/complex/ClientPropertiesTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package org.geotools.data.complex;
```
[the remark this answers is not in the document]
Comment (taba90) [reply modules/extension/app-schema/app-schema/src/test/java/org/geotools/data/complex/ClientPropertiesTest.java:17]: fixed

File: modules/extension/app-schema/app-schema/src/test/resources/test-data/MappedFeatureClientProperties.xml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+				gsml:MappedFeature
+			</targetElement>
+			<attributeMappings>
+				<AttributeMapping>
+					<targetAttribute>
+						gsml:MappedFeature
+					</targetAttribute>
+                    <idExpression>
+                        <OCQL>ID</OCQL>
+                    </idExpression>
```
[the remark this answers is not in the document]
Comment (taba90) [reply modules/extension/app-schema/app-schema/src/test/resources/test-data/MappedFeatureClientProperties.xml]: fixed

File: modules/extension/app-schema/app-schema/src/test/java/org/geotools/data/complex/ClientPropertiesTest.java (hunk 1 above)
Comment (taba90) [reply modules/extension/app-schema/app-schema/src/test/java/org/geotools/data/complex/ClientPropertiesTest.java]: fixed

**Comment by taba90** [comment]
thanks for your review @aaime. Feedbacks applied

**Review by aaime — APPROVED** [review]

**Pull request merged**